nano/node/websocket.cpp

```diff
@@ -4,6 +4,54 @@
 #include <nano/node/node.hpp>
 #include <nano/node/websocket.hpp>
 
+nano::websocket::confirmation_options::confirmation_options (boost::property_tree::ptree const & options_a, nano::node & node_a) :
+all_local_accounts (options_a.get<bool> ("all_local_accounts", false)),
+node (node_a)
+{
+	auto accounts_l (options_a.get_child_optional ("accounts"));
+	if (accounts_l)
+	{
+		for (auto account_l : *accounts_l)
+		{
+			// Check if the account is valid, but no error handling if it's not, simply not added to the filter
+			nano::account result_l (0);
+			if (!result_l.decode_account (account_l.second.data ()))
+			{
+				accounts.insert (account_l.second.data ());
+			}
+		}
+	}
+}
+
+bool nano::websocket::confirmation_options::should_filter (nano::websocket::message const & message_a) const
+{
+	bool should_filter_l (true);
+	auto source_text_l (message_a.contents.get<std::string> ("message.account"));
+	auto destination_opt_l (message_a.contents.get_optional<std::string> ("message.block.link_as_account"));
+	if (!destination_opt_l)
+	{
+		destination_opt_l = message_a.contents.get_optional<std::string> ("message.block.destination");
+	}
+	assert (destination_opt_l);
+	auto destination_text (destination_opt_l.get ());
```

Comment:
destination_opt_l can be not initialized for some old blocks (open, receive, change). So we may add some if else condition or drop legacy blocks support probably

Reply:
Dropping support then. Note for documentation @cryptocode

Reply:
@SergiySW @guilhermelawless I feel this should deliver the exact same blocks as the HTTP callback.

Reply:
Would have been too hard given that we start from an encoded `nano::message`, so i did drop support in the end.

For instance, for receive blocks we would have to get block by its hash.